packages/web/hooks/use-connect-wallet-modal-redirect.tsx

```diff
@@ -1,84 +1,80 @@
-import Image from "next/image";
-import { useState, useEffect, ComponentProps } from "react";
-import { WalletStatus } from "@keplr-wallet/stores";
-import { useStore } from "../../stores";
-import { Button } from "../../components/buttons";
-import { useKeplr } from "./hook";
+import { WalletStatus } from "@cosmos-kit/core";
+import { ComponentProps, useCallback, useEffect, useState } from "react";
 import { t } from "react-multi-lang";
 
+import { Icon } from "~/components/assets";
+
+import { Button } from "../components/buttons";
+import { useStore } from "../stores";
+import { useWalletSelect } from "./wallet-select";
+
 /** FOR USE IN MODALS
  *
  *  If your modal contains a user action that is only possible with a connected wallet on the Osmosis account.
- *  This hook allows your modal to temporarily redirect to the `keplr-connection-selection` modal.
- *  If the user connects Keplr, your modal will appear after. If the user closes the connection
- *  selection modal, your modal will no longer appear, even if they connect Keplr later.
+ *  This hook allows your modal to temporarily redirect to the `wallet-select` modal.
+ *  If the user connects the wallet, your modal will appear after. If the user closes the connection
+ *  selection modal, your modal will no longer appear, even if they connect to the wallet later.
  *
  * @param actionButtonProps Props for the user action `<Button />` that may be replaced with a "Connect Wallet" button which adopts these styles.
  * @param onRequestClose Required callback to request this modal to be popped from `react-modal`. Use to set modal state to `null`/`false`.
  * @returns
  *   * `showModalBase` - Flag for whether the calling component should be shown. `ModalBase` will likely take on these props: `<ModalBase {...props} isOpen={props.isOpen && showModalBase} />`
- *   * `accountActionButton` - JSXElement containing the wrapped action `<Button />` with the given button props (styles). May be a "Connect Account" button if Keplr not connected. Accepts children.
+ *   * `accountActionButton` - JSXElement containing the wrapped action `<Button />` with the given button props (styles). May be a "Connect Account" button if a wallet is not connected. Accepts children.
  */
 export function useConnectWalletModalRedirect(
```

Comment:
I prefer the hook as it gives us more granularity which makes it easier to do minor tweaks if needed. I believe we could improve it by relying on the actual state of the modal instead of the side-effect of when the user pressed the button.